app/src/main/java/fr/neamar/kiss/loader/LoadAppPojos.java

```diff
@@ -18,16 +18,34 @@
 import fr.neamar.kiss.TagsHandler;
 import fr.neamar.kiss.pojo.AppPojo;
 import fr.neamar.kiss.utils.UserHandle;
+import name.pilgr.pipinyin.PiPinyin;
 
 public class LoadAppPojos extends LoadPojos<AppPojo> {
 
     private final TagsHandler tagsHandler;
+    private final PiPinyin piPinyin;
 
     public LoadAppPojos(Context context) {
         super(context, "app://");
+        piPinyin = new PiPinyin(context);
+        
         tagsHandler = KissApplication.getApplication(context).getDataHandler().getTagsHandler();
     }
 
+
+    public static boolean containsHanScript(String s) {
+        for (int i = 0; i < s.length(); ) {
+            int codepoint = s.codePointAt(i);
+            i += Character.charCount(codepoint);
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
```

Comment:
it would probably be more efficient to check for this once at the start of the function, and if not the case you can return false directly, which will speed things up a lot for older devices ;)

Here you check for every character, but the value is guaranteed not to change.